src/cadenza.js

```diff
@@ -432,6 +432,20 @@ export class CadenzaClient {
     return this.#show(resolvePath(backgroundMapView), params, signal);
   }
 
+  /**
+   * Set the visibility of a layer in an embedded map view. The map view must be shown prior to calling this.
+   * For layers in groups, the layer path contains the ancestors also. When making a layer visible, its ancestors will
+   * be made visible also. When hiding a layer, the ancestors are not affected. The operation is asynchronous and its
+   * completion is signaled by an event of type 'setLayerVisibility:success' or 'setLayerVisibility:error'.
+   *
+   * @param {WorkbookLayerPath} layer - The layer to show or hide, identified by its path
+   * @param {boolean} visible - The visibility state of the layer
+   */
+  setLayerVisibility(layer, visible = true) {
+    this.#log('CadenzaClient#setLayerVisibility', ...arguments);
+    this.#postRequest('setLayerVisibility', { layer, visible });
```

Comment:
You also need to return that Promise, otherwise the client can still not know whether it worked.

Reply:
True that